compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -339,6 +339,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             decl_origin: pat_info.decl_origin,
             current_depth: current_depth + 1,
         };
+        let ti = &pat_info.top_info;
```

Comment:
Can we just copy the `TopInfo` everywhere? It doesn't seem *that* large. Also, do you mind inlining this variable to its usages?

Reply:
I inlined the variable :+1: 

`TopInfo` is 40 bytes. Just this morning I was wondering what size is big enough that passing by reference is clearly better for perf than passing by value. I don't know the answer, though 40 is big enough that I start to wonder. There are seven functions in this file that take a `&TopInfo` and ten call sites. I could remove the `&` from every one of them, but I don't feel like the code is obviously better (or worse) that way. Based on all that I'll do the conservative thing and leave that part unchanged.

Reply:
guess so 🤔 seems worth testing anecdotally but i guess this is orthogonal to the cleanup

Reply:
I suspect the answer is the old chestnut "it depends".